### PR TITLE
DEPR: inplace keyword in rename

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx (Replace xxxx with the Github issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Review by datapythonista — COMMENTED** [review]
Looks good to me. Any reason to not add the issue numbers in the PR and the whatsnew, and to not add the appropriate labels to the PR?

File: doc/source/whatsnew/v1.5.0.rst [hunk 1]
```diff
@@ -841,6 +841,7 @@ Other Deprecations
 - Clarified warning from :func:`to_datetime` when delimited dates can't be parsed in accordance to specified ``dayfirst`` argument (:issue:`46210`)
 - Emit warning from :func:`to_datetime` when delimited dates can't be parsed in accordance to specified ``dayfirst`` argument even for dates where leading zero is omitted (e.g. ``31/1/2001``) (:issue:`47880`)
 - Deprecated :class:`Series` and :class:`Resampler` reducers (e.g. ``min``, ``max``, ``sum``, ``mean``) raising a ``NotImplementedError`` when the dtype is non-numric and ``numeric_only=True`` is provided; this will raise a ``TypeError`` in a future version (:issue:`47500`)
+- Deprecated the ``inplace`` keyword in :meth:`DataFrame.rename` and :meth:`Series.rename`, use ``obj = obj.rename(..., copy=False)`` instead (:issue:`??`)
```
Comment (mroeschke) [inline doc/source/whatsnew/v1.5.0.rst:844]:
```suggestion
- Deprecated the ``inplace`` keyword in :meth:`DataFrame.rename` and :meth:`Series.rename`, use ``obj = obj.rename(..., copy=False)`` instead (:issue:`48131`)
```

**Comment by jbrockmendel** [comment]
> Any reason to not add the issue numbers in the PR and the whatsnew, and to not add the appropriate labels to the PR?

No, what issue numbers did you have in mind?

**Comment by datapythonista** [comment]
> No, what issue numbers did you have in mind?

This seemed to me part of #16529, I guess there is not an specific issue for `rename`. That's what I expected to see in the PR description and in the release notes. And I thought having the labels `Deprecate` and `Inplace` was quite obvious too. So, I was wondering if this was a regular PR, or if it had anything special that everything was missing.

**Review by jorisvandenbossche — CHANGES_REQUESTED** [review]
Has there been a recent discussion about `inplace` that we actually want to start doing this deprecation? (the last activity on https://github.com/pandas-dev/pandas/issues/16529 dates from a year ago-
(and as Marc mentions about referring to an issue, that might also help make that clear)

Personally I think we should first discuss how we want to approach this (and ideally someone would actually also first update https://github.com/pandas-dev/pandas/issues/16529 to summarize that discussion and to indicate we want to go forward with it). 
Do we want to do this now? Should we try to do all deprecations / removals in a single cycle (eg that all the keywords could be removed in 2.0, or 3.0) IMO we should try to do that to give a better user experience around this deprecation (it is a _widely_ used keyword), but that also means we should not do that now right before 1.5. 
How does it interact with the other planned copy/view changes?

**Comment by jbrockmendel** [comment]
> Has there been a recent discussion about inplace that we actually want to start doing this deprecation? (the last activity on https://github.com/pandas-dev/pandas/issues/16529 dates from a year ago-

Mostly non-documented discussions on calls from which I've gotten the impression a) some people (e.g. @jreback) feel strongly that inplace should be deprecated across the board, b) no-one is strongly pro-inplace, c) some people (mainly me i think, and I'm wavering) like inplace in the subset of cases where the operation can operate on the underlying data inplace (which is relevant mostly for e.g. fillna, replace, not things like rename).  I'm open to being corrected if this impression is inaccurate.

Also a couple months ago I saw a priorities document (which I can't find now) for the data-apis consortium that included deprecating inplace.  I'm fine with not-prioritizing their priorities if ours are different, but all else equal that makes me more in favor of "get it over with".

> Do we want to do this now? Should we try to do all deprecations / removals in a single cycle (eg that all the keywords could be removed in 2.0, or 3.0)

Users are going to have to update their code for 2.0 and again for 3.0 regardless of what we do with inplace.  I'm not inclined to worry about this.  The constraint is developer time.

**Comment by jorisvandenbossche** [comment]
> Also a couple months ago I saw a priorities document (which I can't find now) for the data-apis consortium that included deprecating inplace.

Just to be clear here (although I was involved myself): that's a document of a private meeting, and thus has no value for decision making in pandas without that those issues are also discussed on the public pandas channels. It can help for people involved there to prioritize their work, but when actually starting with that, they still need to go through the normal procedures here.

I think the impression that most core devs seem to be onboard with this change is correct (that's also my impression), but I think a good summary of the discussion in https://github.com/pandas-dev/pandas/issues/16529 would still be useful (especially because this is a change that affects _a lot_ of users, we have to communicate this change and the reasoning very well)

**Pull request closed without merging**